demos/src/mqtt_demo_plaintext.c

```diff
@@ -168,19 +168,52 @@ static int establishMqttSession( MQTTContext_t * pContext, int tcpSocket )
     return status;
 }
 
+static int disconnectMqttSession( MQTTContext_t * pContext )
+{
+    int status = EXIT_SUCCESS;
+
+    MQTTStatus_t mqttStatus = MQTT_Disconnect( pContext );
+
+    if( mqttStatus != MQTTSuccess )
+    {
+        status = EXIT_FAILURE;
+    }
+
+    return status;
+}
+
 int main( int argc, char ** argv )
 {
+    bool mqttSessionEstablished = false;
     int status;
     MQTTContext_t context;
+
     int tcpSocket = connectToServer( SERVER, PORT );
 
-    if( tcpSocket != -1 )
+    if( tcpSocket == -1 )
     {
-        status = establishMqttSession( &context, tcpSocket );
+        status = EXIT_FAILURE;
     }
     else
     {
-        status = EXIT_FAILURE;
+        status = establishMqttSession( &context, tcpSocket );
+
+        if( status == EXIT_SUCCESS )
+        {
+            mqttSessionEstablished = true;
+        }
+    }
+
+    if( mqttSessionEstablished == true )
+    {
+        if( status == EXIT_FAILURE )
```

Comment:
Both the if-else are doing the same thing - Should we not just call the disconnectMqttSession and ignore the return code? In fact should ```disconnectMqttSession``` even have a return code?

Reply:
The return code from disconnectMqttSession comes from MQTT_Disconnect.  This demo checks the return code only as a "best practice", though I agree it's probably not necessary.  I can remove it if you think we should.